tests/subsys/pm/device_wakeup_api/src/main.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <device.h>
+#include <pm/pm.h>
+
+#define DEV_NAME DT_LABEL(DT_ALIAS(gpiotest))
+
+
+static const struct device *dev;
+static uint8_t sleep_count;
+
+
+void pm_power_state_set(struct pm_state_info info)
+{
+	ARG_UNUSED(info);
+
+	enum pm_device_state state;
+
+	switch (sleep_count) {
+	case 1:
+		/* Just  a sanity check that the system is the right state.
+		 * Devices are suspended before SoC on PM_STATE_SUSPEND_TO_RAM, that is why
+		 * we can check the device state here.
+		 */
+		zassert_equal(info.state, PM_STATE_SUSPEND_TO_RAM, "Wrong system state");
+
+		(void)pm_device_state_get(dev, &state);
+		zassert_equal(state, PM_DEVICE_STATE_SUSPENDED, "Wrong device state");
+
+		/* Enable wakeup source. Next time the system is called
+		 * to sleep, this device will still be active.
+		 */
+		(void)pm_device_wakeup_enable((struct device *)dev, true);
+		break;
+	case 2:
+		zassert_equal(info.state, PM_STATE_SUSPEND_TO_RAM, "Wrong system state");
+
+		/* Second time this function is called, the system is asked to standby
+		 * and devices were suspended.
+		 */
+		(void)pm_device_state_get(dev, &state);
+		zassert_equal(state, PM_DEVICE_STATE_ACTIVE, "Wrong device state");
+		break;
+	default:
+		break;
+	}
+}
+
+void pm_power_state_exit_post_ops(struct pm_state_info info)
+{
+	irq_unlock(0);
+}
+
+struct pm_state_info pm_policy_next_state(int32_t ticks)
+{
+	while (sleep_count < 3) {
+		sleep_count++;
+		return (struct pm_state_info){PM_STATE_SUSPEND_TO_RAM, 0, 0, 0};
+	}
+
+	return (struct pm_state_info){PM_STATE_ACTIVE, 0, 0, 0};
+}
+
+void test_wakeup_device_api(void)
+{
+	bool ret = false;
+
+	dev = device_get_binding(DEV_NAME);
```

Comment:
Just use DEVICE_DT_GET. There's no reason anymore to be using device_get_binding for devices that are always set up in devicetree.